js/whitebit.js

```diff
@@ -974,64 +974,65 @@ module.exports = class whitebit extends Exchange {
+                if (isLimitOrder) {
+                    // stop limit order
+                    method = 'v4PrivatePostOrderStopLimit';
+                    request['price'] = this.priceToPrecision (symbol, price);
+                } else {
+                    // stop market order
+                    method = 'v4PrivatePostOrderStopMarket';
+                }
+            } else {
+                if (isLimitOrder) {
+                    // limit order
+                    method = 'v4PrivatePostOrderNew';
+                    request['price'] = this.priceToPrecision (symbol, price);
+                } else {
+                    // market order
+                    method = 'v4PrivatePostOrderMarket';
+                }
             }
-            const convertedPrice = this.priceToPrecision (symbol, price);
-            request['price'] = convertedPrice;
-        }
-        if (type === 'market' || type === 'stopMarket') {
-            if (side === 'buy') {
-                let cost = this.safeNumber (params, 'cost');
+            if (isMarketOrder && side === 'buy') {
+                let cost = undefined;
                 const createMarketBuyOrderRequiresPrice = this.safeValue (this.options, 'createMarketBuyOrderRequiresPrice', true);
                 if (createMarketBuyOrderRequiresPrice) {
-                    if (price !== undefined) {
-                        if (cost === undefined) {
-                            cost = amount * price;
-                        }
-                    } else if (cost === undefined) {
-                        throw new InvalidOrder (this.id + " createOrder() requires the price argument for market buy orders to calculate total order cost (amount to spend), where cost = amount * price. Supply a price argument to createOrder() call if you want the cost to be calculated for you from price and amount, or, alternatively, add .options['createMarketBuyOrderRequiresPrice'] = false and supply the total cost value in the 'amount' argument or in the 'cost' extra parameter (the exchange-specific behaviour)");
+                    if (price === undefined) {
+                        throw new InvalidOrder (this.id + " createOrder () requires the price argument with market buy orders to calculate total order cost (amount to spend), where cost = amount * price. Supply a price argument to createOrder() call if you want the cost to be calculated for you from price and amount, or, alternatively, add .options['createMarketBuyOrderRequiresPrice'] = false and supply the total cost value in the 'amount' argument");
                     }
+                    cost = amount * price;
                 } else {
-                    cost = (cost === undefined) ? amount : cost;
+                    cost = amount;
                 }
-                request['amount'] = this.costToPrecision (symbol, cost);
+                request['amount'] = this.priceToPrecision (symbol, cost);
```

Comment:
@DoctorSlimm, why are we using `priceToPrecision` instead of `costToPrecision` here?

Reply:
actually that looks like it should be `amountToPrecision` )

Reply:
@frosty00 nop because this is a `createMarketBuyOrderRequiresPrice` case 🙂